nixos/modules/services/web-apps/mediawiki.nix

```
@@ -503,6 +533,62 @@ in
         }
       ];
     };
+    # inspired by https://www.mediawiki.org/wiki/Manual:Short_URL/Nginx
+    services.nginx = lib.mkIf (cfg.webserver == "nginx") {
+      enable = true;
+      virtualHosts.${config.services.mediawiki.nginx.hostName} = {
+        root = "${pkg}/share/mediawiki";
+        locations = {
+          "~ ^/w/(index|load|api|thumb|opensearch_desc|rest|img_auth)\\.php$".extraConfig = ''
+            rewrite ^/w/(.*) /$1 break;
+            include ${config.services.nginx.package}/conf/fastcgi_params;
+            fastcgi_index index.php;
+            fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+            fastcgi_pass unix:${config.services.phpfpm.pools.mediawiki.socket};
+          '';
+          "/w/images/".alias = withTrailingSlash cfg.uploadsDir;
+          # Deny access to deleted images folder
+          "/w/images/deleted".extraConfig = ''
+            deny all;
+          '';
+          # MediaWiki assets (usually images)
+          "~ ^/w/resources/(assets|lib|src)" = {
+            tryFiles = "$uri =404";
+            extraConfig = ''
```

This should have been

```suggestion
            extraConfig = ''
              rewrite ^/w(/.*) $1 break;
```

---

Can you make a pull request?

---

Or if not, how can I test these changes?

---

[These](https://trong.loang.net/nixos-conf/commit/?id=5f1653842dd91fbeff94472f6e3848f12492fd9f) are deployed on our wiki instance, e.g. [/w/skins/Vector/resources/common/images/link-external-small-ltr-progressive.svg](https://nhanh.loang.net/w/skins/Vector/resources/common/images/link-external-small-ltr-progressive.svg) now loads.  The same fix doesn't work for `/w/rest.php` though so I'm hesitant to submit the incomplete patch.